sklearn/manifold/spectral_embedding_.py

```diff
@@ -282,6 +282,8 @@ def spectral_embedding(adjacency, n_components=8, eigen_solver=None,
             laplacian *= -1
 
     if eigen_solver == 'amg':
+        if eigen_tol == 0:
```

Comment:
How about `eigen_tol = max(1e-12, eigen_tol)` to handle float comparison issues?